src/app/(post-login)/submission/program/[shortName]/clinical-data/ClinicalEntityDataTable/index.tsx

```diff
@@ -0,0 +1,636 @@
+							const isSticky = value === 'donor_id';
+							const isSorted = props.sorted;
+
+							return <Cell config={{ isLastElement, isSorted, isSticky }}>{value}</Cell>;
+						},
+						meta: { customCell: true, customHeader: true },
+						cell: (context) => {
+							const value = context.getValue();
+							const isSticky = context.column.id === 'donor_id';
+
+							const { isCompletionCell, errorState, style } = getCellStyles(
+								undefined,
+								context.row,
+								context.column,
+							);
+
+							const showSuccessSvg = isCompletionCell && !errorState;
+							const { hasMissingEntityException } = context.row.original;
+							const colId = context.column.id;
+							const showMissingEntitySymbol =
+								hasMissingEntityException &&
+								[completionColumnHeaders.treatments, completionColumnHeaders.followUps].includes(
+									colId,
+								);
+
+							const content = showMissingEntitySymbol ? (
+								<div>{DashIcon}</div>
+							) : showSuccessSvg ? (
+								<Icon name="checkmark" fill="accent1_dimmed" width="12px" height="12px" />
+							) : (
+								value
+							);
+
+							return (
+								<Cell config={{ isSticky }} styles={[style]}>
+									{content}
+								</Cell>
+							);
+						},
+					})),
+				},
+				{
+					id: 'submitted_donor_data_header',
+					meta: { customHeader: true },
+					header: () => (
+						<TopLevelHeader
+							title="SUBMITTED DONOR DATA"
+							styles={[
+								css`
+									text-align: left;
+								`,
+							]}
+						/>
+					),
+					columns: columns.slice(7),
+				},
+			];
+		}
+
+		const tableMin = totalDocs > 0 ? page * pageSize + 1 : totalDocs;
+		const tableMax = totalDocs < (page + 1) * pageSize ? totalDocs : (page + 1) * pageSize;
+		const numTablePages = Math.ceil(totalDocs / pageSize);
+		const numErrorPages = Math.ceil(totalErrorCount / errorPageSize);
+
+		return (
+			<div
+				ref={containerRef}
+				css={css`
+					position: relative;
+				`}
+			>
+				{hasErrors && (
+					<div
+						id="error-submission-workspace"
+						css={css`
+							margin: 12px 0px;
+						`}
+					>
+						<ErrorNotification
+							level={NOTIFICATION_VARIANTS.ERROR}
+							title={`${totalErrorCount.toLocaleString()} error(s) found on the current page of ${clinicalEntityDisplayNames[
+								entityType
+							].toLowerCase()} table`}
+							subtitle={<Subtitle program={program} />}
+							reportData={tableErrors}
+							reportColumns={errorColumns}
+							tableProps={{
+								page: errorPage,
+								pages: numErrorPages,
+								pageSize: errorPageSize,
+								sorted: errorSorted,
+								onPageChange: (value) => updatePageSettings('page', value),
+								onPageSizeChange: (value) => updatePageSettings('pageSize', value),
+								onSortedChange: (value) => updatePageSettings('sorted', value),
+								// TODO: Test + Update Pagination in #2267
+								// https://github.com/icgc-argo/platform-ui/issues/2267
```

Comment:
This Todo can be removed

Reply:
thank you - the exact reason I'm against putting TODOs in code